crates/services/p2p/src/service.rs

```diff
@@ -115,6 +115,10 @@ pub enum TaskRequest {
         channel: OnResponse<Option<Vec<SealedBlockHeader>>>,
     },
     GetTransactions {
+        block_height_range: Range<u32>,
+        channel: OnResponse<Option<Vec<Transactions>>>,
+    },
+    GetTransactionsFromPeer {
```

Comment:
I think it's useful in case of syncing when you already received the header from someone in the current syncing process and so you can directly ask him without having to redo peer selection. Maybe there is also othhers use cases but this is the only one I see.

Reply:
It seems that after your PR we don't need this variant anymore. But we can address it in your PR